### ISSUE TITLE
Fail to build project

### ISSUE THREAD

**Issue by eecheng87** [3541/short-circuit#7]
Hello, thanks for your interesting project.

Following the guide, I create a build directory and download the submodule but get the following error message after typing `meson setup <BUILDDIR>`, anything I miss here?

```
|Run-time dependency highwayhash_c found: NO (tried pkgconfig and cmake)
|Looking for a fallback subproject for the dependency highwayhash_c
|subprojects/a3/meson.build:112: WARNING: Dependency highwayhash not found but it is available in a sub-subproject.
|To use it in the current project, promote it by going in the project source
|root and issuing the following command: 
|meson wrap promote subprojects/a3/subprojects/highwayhash.wrap

subprojects/a3/meson.build:112:0: ERROR: Subproject directory not found and highwayhash.wrap file not found
```

Thanks in advance!

**Comment by 3541** [comment]
This is a result of Meson being pedantic about dependency resolution and refusing to automatically pull in transitive dependencies (`sc` depends on `a3` depends on HighwayHash). You can run the command suggested in the error message (`meson wrap promote subprojects/a3/subprojects/highwayhash.wrap`), and it should fix the issue. I believe newer versions of Meson do not have this issue.

**Comment by eecheng87** [comment]
Hi, thanks for your quick reply. You're right, newer versions of Meson do not have this issue.

**Comment by eli-schwartz** [comment]
> I believe newer versions of Meson do not have this issue.

Since 0.56 Meson does auto-promotion (unless you configure with --wrap-mode=nopromote).

**Comment by 3541** [comment]
> Since 0.56 Meson does auto-promotion (unless you configure with --wrap-mode=nopromote).

Ah, nice to know the exact version. I will update the README.

**Comment by eli-schwartz** [comment]
No problem.

This is documented at https://mesonbuild.com/Subprojects.html#commandline-options but it might be a bit obscure, since it's a note about --wrap-mode=nopromote, pointing out that *that* option got added in 0.56.0 in order to manually disable behavior introduced in the same version.

Anyway, we've reached the point where Debian stable is good enough for this and people with older versions are becoming increasingly uncommon. :)
